### DISCUSSION TITLE
How to handle load balancing IP?

### DISCUSSION THREAD

**Discussion by Mercusio** [netbox-community/netbox#9271] [Q&A] [answered]
Hello,

I am pretty new to netbox, but I have search quite a lot on this subject and can't seem to find how to do it.
How can I handle load balancing IP addresses?

Let's say I have one VMWare host with multiple interfaces. 2 of them are using the same IP address, both on different switches.
That way if one of them fails, the host can still communicate.

It seemed to me that FHRP Groups were meant for this, but I can't choose an FHRP Group to assign on an interface.

Anyone have any idea?

Thanks!

**Comment by Houtek** [comment]
Hi,
we use the IP role "VIP" for this kind of IPs on loadbalancers and HSRP addresses.
I believe, I read it somewhere on Slack or so long time ago.

**Reply by Mercusio** [reply]
> Hi, we use the IP role "VIP" for this kind of IPs on loadbalancers and HSRP addresses. I believe, I read it somewhere on Slack or so long time ago.

Hi,
Thanks for taking the time to answer.
Unfortunately this does not allow me to assign the same IP address to 2 interfaces.

**Comment by candlerb** [comment, marked as answer]
> but I can't choose an FHRP Group to assign on an interface

The user interface to reassign an IP address to an FHRP group (as opposed to adding a new IP address to an FHRP group) is very awkward: see #8106.  The section "Use case" describes the sequence of steps currently required.

To answer your specific question about how to "choose an FHRP Group to assign on an interface": you have to *view* the interface, by clicking the interface name in the left hand column (not *edit* it by clicking the pencil icon).

If you do this, then on the right hand side of the interface details page you get a panel which allows you to assign FHRP groups to the interface:
![image](https://user-images.githubusercontent.com/44789/166502754-88d5a449-36ef-4bec-801b-0efad57d3116.png)

It definitely was not obvious to me (my logic was: surely, to modify the list of FHRP groups on an interface, you have to *edit* the interface? No!)

**Reply by Mercusio** [reply]
Thank you, this works. This is what I was looking for.
It's a shame you can't see the IP address next to the interface on the device page though.

**Reply by bluikko** [reply]
FHRP does not really sound like the right solution. Literally there is a protocol behind FHRP for floating IP addresses and your example case doesn't sound like that - though it is not exactly clear to me, does the example mean for example VMware management IP address?

**Reply by candlerb** [reply]
I use FHRP groups for "floating" addresses that can move between hosts. It doesn't really matter whether this movement is automatic or manual. FHRP protocol "Other" (#8100) covers the manual case.

A cluster of hosts, where the management IP address could be assigned to any host but only one of them at a given instant in time, is a good match for this IMO.

**Reply by bluikko** [reply]
Yes I agree. But the OP was about a single host and multiple interfaces.

**Reply by candlerb** [reply]
Oh yes, you're right. For that case I would create a virtual interface for the IP address, since it is reachable (at layer 2) through multiple physical interfaces.

**Reply by Mercusio** [reply]
> Oh yes, you're right. For that case I would create a virtual interface for the IP address, since it is reachable (at layer 2) through multiple physical interfaces.

You are absolutely right. I've thought too much about this and looked for complicated solution when it was in fact very simple.
A virtual IP is the way to go, thank you.
